src/modules/drawing/components/useDrawingModeInteraction.composable.js

```diff
@@ -146,49 +194,54 @@ export default function useDrawingModeInteraction({
-        editableFeature.geometry = new GeoJSON().writeGeometryObject(geometry)
-
-        // removing the flag we've set above in onDrawStart (this feature is now drawn)
-        feature.unset('isDrawing')
-        // setting the definitive style function for this feature (thus replacing the editing style from the interaction)
-        // This function will be automatically recalled every time the feature object is modified or rerendered.
-        // (so there is no need to recall setstyle after modifying an extended property)
-        feature.setStyle(geoadminStyleFunction)
-        // see https://openlayers.org/en/latest/apidoc/module-ol_interaction_Draw-Draw.html#finishDrawing
-        interaction.finishDrawing()
-        store.dispatch('addDrawingFeature', { featureId: feature.getId(), ...dispatcher })
-        store.dispatch('setDrawingMode', { mode: null, ...dispatcher })
-        if (drawEndCallback) {
-            drawEndCallback(feature)
+        if (isExtending) {
+            updateStartingFeature(drawnFeature)
+        } else {
+            // deactivating the interaction (so that the user doesn't create another feature right after this one)
+            // this does not change the state, for that we will bubble the event so that the parent will then
+            // dispatch changes to the store
+            // deactivate()
+            // grabbing the drawn feature so that we send it through the event
+
+            const editableFeature = drawnFeature.get('editableFeature')
+            editableFeature.setCoordinatesFromFeature(drawnFeature)
+            // setting the geometry too so that the floating popup can be placed correctly on the map
+            editableFeature.geometry = new GeoJSON().writeGeometryObject(geometry)
+
+            // removing the flag we've set above in onDrawStart (this feature is now drawn)
+            drawnFeature.unset('isDrawing')
+            // setting the definitive style function for this feature (thus replacing the editing style from the interaction)
+            // This function will be automatically recalled every time the feature object is modified or rerendered.
+            // (so there is no need to recall setstyle after modifying an extended property)
+            drawnFeature.setStyle(geoadminStyleFunction)
+            // see https://openlayers.org/en/latest/apidoc/module-ol_interaction_Draw-Draw.html#finishDrawing
+            interaction.finishDrawing()
+            store.dispatch('addDrawingFeature', { featureId: drawnFeature.getId(), ...dispatcher })
+            store.dispatch('setDrawingMode', { mode: null, ...dispatcher })
+            if (drawEndCallback) {
+                drawEndCallback(drawnFeature)
+            }
+            // Here we need to save work in next tick to have the drawingLayer source updated.
+            // Otherwise, the source might not yet be updated with the new/updated/deleted feature
+            nextTick().then(debounceSaveDrawing)
```

Comment:
would a watch (deep watch if necessary) on the drawing layer also solve this issue?
Here you are waiting an arbitrary life cycle before committing to saving, but we can't be sure the next tick will be the one that has saved/stored everything you need, isn't it?

Reply:
It's not my original code actually. It's already there in develop branch, see [this](https://github.com/geoadmin/web-mapviewer/blob/develop/src/modules/drawing/components/useDrawingModeInteraction.composable.js#L189-L191). I only did a small refactor, i.e. put it in a if statement. So, I am not really sure what to do.

here is the original PR that introduce the code: https://github.com/geoadmin/web-mapviewer/pull/561/files#diff-3423e4552122109b2a6f9dac47e5d135482d4822ce8a85c113ec2c983463a51cR170-R172